agent_identity/src/service/aggregate.rs

```diff
@@ -16,15 +12,22 @@ use identity_credential::{
     credential::Jwt,
     domain_linkage::{DomainLinkageConfiguration, DomainLinkageCredentialBuilder},
 };
-use identity_did::{CoreDID, DIDUrl};
+use identity_did::DIDUrl;
 use identity_document::service::{Service as DocumentService, ServiceEndpoint};
-use jsonwebtoken::Header;
-use oid4vc_core::authentication::subject::Subject as _;
+use jsonwebtoken::{Algorithm, Header};
+use oid4vc_core::Sign as _;
 use serde::{Deserialize, Serialize};
 use serde_json::json;
-use std::sync::Arc;
+use std::{str::FromStr as _, sync::Arc};
 use tracing::info;
 
+#[derive(Debug, Clone, Serialize, Deserialize, Default, PartialEq)]
+pub enum Status {
+    #[default]
+    Created,
```

Comment:
I don't fully understand the requirement for such a status here. If you want to offer to the user that it can be reactivated, shouldn't it be "active" and "inactive"? If this is purely about tracking when it should not be displayed anymore (due to the immutability of the events), a simple "is_deleted" flag should be enough, right?

Reply:
True, I guess I was thinking since multiple Aggregates now have a status field I wanted to be consistent in that sense. But you're right that an `is_deleted` flag makes more sense. 

A `status` field included in an Aggregate should only have a 'semantic' value specific to the aggregate (which is indeed not the case here). :+1: